### ISSUE TITLE
RHEL support

### ISSUE THREAD

**Issue by willwilliams** [facebook/fbcunn#29]
Hi,

I'm running centos and there's a fair bit of adaptation required to get the install script working. I've not managed it yet. Does anyone have a list of the dependencies required to install this on an RHEL based machine?

It's great that you support Ubuntu - would be even better if the install covered the other majority of linux users that aren't debian based. Are there any plans?

Thanks!

**Comment by soumith** [comment]
fbcunn unfortunately wont work on RHEL6 and below. RHEL does not have a modern GCC that is needed. Did not try it in RHEL7. 

If you want to use specific parts of fbcunn like DataParallel and ModelParallel, i am isolating them out to a pure-lua package with no dependencies.

**Comment by willwilliams** [comment]
That's great - it was OneBitDataParallel that I was really wanting to try out so I hope those make it into the separate lua package aswell.

**Comment by hohoCode** [comment]
Yes that will be great! I had same question too (I guess many others have..). .... And could you please also isolate that fast LookupTable and fast fft temporal convolutional layer in fbcunnn too?... Super much.. needed. Thank you. Appreciate it.

**Comment by soumith** [comment]
I cannot isolate anything that touches the C++ or CUDA code. So Neither the fast LookupTable or the OneBitDataParallel (which uses CUDA code for quantize/unquantize) make sense to isolate. I will isolate DataParallel and ModelParallel

**Comment by soumith** [comment]
if you all want to run the imagenet multi-GPU example on RHEL or OSX, here is a fork that does not need fbcunn: https://github.com/soumith/imagenet-multiGPU.torch

**Comment by facebook-github-bot-5** [comment]
Thank you for reporting this issue and appreciate your patience. We've  notified the core team for an update on this issue. We're looking for a response within the next 30 days or the issue may be closed.
